### DISCUSSION TITLE
Changing mania skin during pause leaveing weird artifact

### DISCUSSION THREAD

**Discussion by GSculerlor** [ppy/osu#14993] [Q&A] [answered]
Creating a discussion instead of issue to make sure it's not a duplicate or a known issue.

step to reproduce: 
- play a beatmap on mania ruleset
- pause during gameplay
- change the skin

video:

https://user-images.githubusercontent.com/6506864/136378634-12582931-3f8b-41c2-aa48-0d638250e853.mp4

logs:
[performance.log](https://github.com/ppy/osu/files/7302869/performance.log)
[runtime.log](https://github.com/ppy/osu/files/7302866/runtime.log)

edit: upon closer look at other ruleset seems like this weird behavior also happens

**Comment by smoogipoo** [comment, marked as answer]
For anyone following along, this is because updates are blocked while gameplay is paused so `FillFlowContainer`s can't reflow, but some elements bind to skin changes to alter size and the new size is reflected in draw nodes.

Example:
https://github.com/ppy/osu/blob/f8e50731b733df347c0df19aaf483d330f4e2674/osu.Game.Rulesets.Mania/UI/ColumnFlow.cs#L50-L82

Perhaps it's possible to not bind in such a way, and instead create new UI components. But I wouldn't say this is a priority.

**Reply by peppy** [reply]
If anything, I think our goal should be to actually allow the interface to update, rather than blocking, but that's going to also require some thinking.
